### PR TITLE
Fix flatten list function to handle column with lists in it

### PR DESCRIPTION
**Pull request by nhaydel**

### CODE REVIEW

**Review by mdeforestbrown — APPROVED** [review]
Thanks for adding this, could we add a test to evidence how this was failing so I can understand better what I missed?

**Comment by nhaydel** [comment]
> Thanks for adding this, could we add a test to evidence how this was failing so I can understand better what I missed?

Sure thing, I will add a test for the flatten_list function so it is more clear the expected output of that function

**Comment by nhaydel** [comment]
@mdeforestbrown I have added tests for flatten list. I'm not sure why the test action fails in github, but I think that is something that can be resolved in another issue. The tests pass for me locally

File: business_rules/operators.py [hunk 1]
```diff
@@ -860,13 +860,14 @@ def is_unique_set(self, other_value):
         target = self.replace_prefix(other_value.get("target"))
         comparator = other_value.get("comparator")
         values = [target, comparator]
-        target_data = flatten_list(values)
+        target_data = flatten_list(self.value, values)
```
Comment (mdeforestbrown) [inline business_rules/operators.py:863]: What's an example of `self.value` in this instance?

Comment (mdeforestbrown) [reply business_rules/operators.py:863]: @nhaydel  Okay I've gone back through this, so am I correct in saying that the use case I missed is below.

```
self.value = {
               "ARM": [["PLACEBO", "PLACEBO", "A", "A"], ["PLACEBO", "PLACEBO", "A", "A"]],
                "TAE": [1, 1, 1, 2],
                "LAE": [1, 2, 1, 2],
                "ARF": [1, 2, 3, 4],
            }
l = ["ARM", "LAE"]
```

If yes, is it worth me adding that case to the `test_is_unique_set` unit tests?

Comment (nhaydel) [reply business_rules/operators.py:863]: I think so if they don't have tests already. And yes, self.value is a value of the type the operator is implemented for. For the dataframe type self.value is the pandas.Dataframe the rule is operating on

**Pull request merged**